### ISSUE TITLE
Customize reception of push notification

### ISSUE THREAD

**Issue by ricardoribas** [OneSignal/react-native-onesignal#952]
**Description:**

There is any way to customize the logic to determine whether or not we should receive a push notification? Imagine a use case in which a user is using an app that has chat groups. If we are inside a chat group we don't want to receive push notifications, but we should receive push notifications related to other groups.

The [official documentation](https://documentation.onesignal.com/v5.0/docs/react-native-sdk#section-subscription-status) has a property that might help on this topic, but it might not help if we are dealing with multiple chats.

> **Subscription Status**
Change User Subscription Status
Use this method only on subscribed users. It will not subscribe users that have turned off push in the App Settings or denied subscription through the native prompt on iOS.
You can call this method with false to opt users out of receiving all notifications through OneSignal. You can then pass true later to opt users back into notifications
You cannot resubscribe users that have opted out of push in any other way besides setting 
`setSubscription(false);`

Some help needed here.

**Comment by rborn** [comment]
@ricardoribas I tihnk is a matter of setting `kOSSettingsKeyInFocusDisplayOption` to `none`  https://documentation.onesignal.com/docs/react-native-sdk#section-i-os-initialization-parameters  and then build and show the notification yourself (will be a normal View inside the app) depending on the chat room you are in.

**Comment by ricardoribas** [comment]
I think that will not solve my problem as i don't want to receive push notifications at all for that specific chat group. I manage to solve this issue by tagging users that are inside a chat group. However, this introduces another question:
If i define a list of devices **[device_id_A, device_id_B]**  and a filter for the tag 
`
{
        field: 'tag',
        key: MY_TAG,
        relation: '!=',
        value: my_value
      }
`
it will filter the list of devices that i pass or the entire list of subscribed users?

**Comment by ricardoribas** [comment]
I saw a note in the SDK reference:

> The Create Notification method is used when you want your server to programmatically send notifications or emails to a segment or individual users. You may target users in one of three ways using this method: by Segment, by Filter, or by Device. At least one targeting parameter must be specified.

I wonder if there is any workaround for the use case i would like to achieve.

**Comment by rborn** [comment]
The tag filters all the users belonging to that tag, or at least this is my understanding.

**Comment by ricardoribas** [comment]
But if you use filters, the list of devices whereas you want to send the push notifications will be ignored. Is what I want to want to achieve. I want users that DO NOT have the tag and have the devices.

**Comment by rborn** [comment]
I think you need help from somebody from OneSignal 😅

**Comment by ricardoribas** [comment]
> I think you need help from somebody from OneSignal

Indeed. Thanks your all your help. Appreciate it.

**Comment by rgomezp** [comment]
Yes, you should add tags for this use case. The notification will be sent to all users who have the tag you specify. For example, if you want to target users in a chat group called "chatgroup1", you would use

`{ field: 'group', key: MY_TAG, relation: '=', value: 'chatgroup1' }`

Can you rephrase your question please? What do you mean that you want to target users that don't have the tag **and have the devices**?

**Comment by rgomezp** [comment]
Closing due to no response
